File: API-Angular/Models/StudentCRUDDTOs/StudentDTO.cs
using API_Angular.Models.StudentCRUD;

namespace API_Angular.Models.StudentCRUDDTOs;

public class StudentDTO
{
    public int StudentId { get; set; }
    public string Name { get; set; }
    public string DateOfBirth { get; set; }
    public bool Passed { get; set; } // checkbox
    public string Gender { get; set; } // radio

    public int CountryId { get; set; }

    //public List<Subject> Subjects { get; set; } // sk..


}

File: Client-Angular/src/app/Models/StudentDTO.ts
import { Country } from "../country";
import { Subject } from "../subject";

export class StudentDTO {

  studentId: number | any;
  name: string | any;
  dateOfBirth: string | any;
  passed: boolean | any;  // checkbox ... act
  gender: string | any;   // radio ... sta

  countryId: number | any;    // cli...
  country: Country;     // cli...

  subjects: Subject[];

  constructor() {
    this.studentId = null;
    this.name = null;
    this.dateOfBirth = null;
    this.passed = true;
    this.gender = null;

    this.countryId = null;
    this.country = new Country();

    this.subjects = [];
  }

}
